src/Chainweb/Pact/PactService.hs

```diff
@@ -1012,17 +1014,13 @@ playOneBlock currHeader plData pdbenv = do
     cp <- getCheckpointer
     let creationTime = _blockCreationTime currHeader
     -- prop_tx_ttl_validate
-    oks <- liftIO $ fmap (either (const False) (const True)) <$>
-        validateChainwebTxs cp creationTime (_blockHeight currHeader) trans skipDebitGas
-
-    let mbad = V.elemIndex False oks
-    case mbad of
-        Nothing -> return ()  -- ok
-        Just idx -> let badtx = (V.!) trans idx
-                        hash = P._cmdHash badtx
-                        msg = [ (hash, validationErr hash) ]
-                    in throwM $ TransactionValidationException msg
-    -- transactions are now successfully validated.
+    valids <- V.zip trans <$> liftIO
+      (validateChainwebTxs cp creationTime (_blockHeight currHeader) trans skipDebitGas)
+
+    case foldr handleValids [] valids of
+      [] -> return ()
+      errs -> throwM $ TransactionValidationException $ errs
+
```

Comment:
This shouldn't break anything, wanted to see the actual errors when validation fails. The errors blow up and don't end up in payload outputs.